### PR TITLE
Move Sitemaps to Django Sitemap Framework

### PR DESCRIPTION
**Pull request by saadmk11**
closes https://github.com/djangopackages/djangopackages/issues/1146

**Note:** The Site Object (http://127.0.0.1:8000/admin/sites/site/) needs to be updated with proper domain name. i.e: `djangopackages.org`

### CODE REVIEW

**Review by ryancheley — APPROVED** [review]
Looks great! 💯

File: package/sitemaps.py
Comment (jefftriplett) [inline package/sitemaps.py:7]:
```suggestion
    priority = 1.0
    protocol = "https"
```

File: grid/sitemaps.py
Comment (jefftriplett) [inline grid/sitemaps.py:7]:
```suggestion
    priority = 1.0
    protocol = "https"
```

File: urls.py
Comment (jefftriplett) [inline urls.py:32]: Let's drop this to 12 hours and see what it looks like. I have been bitten by 24 caching + once-a-day check, meaning ~2-day lag. 

```suggestion
SITEMAPS_CACHE_TTL = 12 * 60 * 60  # 24 Hours
```

**Review by jefftriplett — CHANGES_REQUESTED** [review]
This looks great! I had only one nitpick on the protocol but I left suggestions for everything else. I'm happy to merge and deploy this after those get merged in. Good work!

File: grid/sitemaps.py
Comment (ryancheley) [reply grid/sitemaps.py:7]: Since we’re on Django 5 do we need to be explicit with the protocol? Per the [docs](https://docs.djangoproject.com/en/5.0/ref/contrib/sitemaps/#django.contrib.sitemaps.Sitemap.protocol)

> Changed in Django 5.0:
> In older versions, the default protocol for sitemaps built outside the context of a request was 'http'.

Comment (saadmk11) [reply grid/sitemaps.py:7]: @jefftriplett I intensionally left it as is. Because:

> If protocol is None, the URLs in the sitemap will use the protocol with which the sitemap was requested.

This will make the views generate proper URLs both in production and in development. (Assuming we don't allow http on production)

Ref: https://github.com/django/django/blob/8c257cecffc314179a5375ab8a16ca0ff3b6fb16/django/contrib/sitemaps/__init__.py#L13

Comment (jefftriplett) [reply grid/sitemaps.py:7]: Perfect. After ~18 years of it not working, I'm thrilled to learn that it finally "just works"

**Review by jefftriplett — APPROVED** [review]

**Pull request merged**
